translib/db/db_export.go

```diff
@@ -141,7 +141,8 @@ func generateDbDump(filePath, rawDumpFile string) (string, error) {
 func createFile(template string) (*os.File, error) {
 	dirname, basename := filepath.Split(template)
 	if len(basename) == 0 || strings.IndexByte(basename, '*') >= 0 {
-		return os.CreateTemp(dirname, basename)
+		//TODO change to os.CreateTemp after upgrading to the latest golang
+		return ioutil.TempFile(dirname, basename)
```

Comment:
As of go 1.17, this function gets deprecated. Can we upgrade go version instead?

Reply:
Upgrading golang version is not trivial -- there is also a FIPS dependency. Moreover it will automatically be taken care as part of the debian version upgrade activity. Did you notice https://github.com/sonic-net/sonic-buildimage/blob/master/sonic-slave-bookworm/Dockerfile.j2#L451?? The bookworm docker uses go1.19 for normal build and go1.15 for fips builds (i don't know why it was done that way). So, the code is expected to work with both versions. Hence changing `os.CrateTemp()` calls to `ioutil.TempFile()` is the correct fix for now.. I understand this API has been deprecated, but the functionality has been retained. The latest implementation simply calls `os.CreateTemp()`.